sos/report/plugins/microshift.py

```diff
@@ -0,0 +1,162 @@
+        allow for end users to specify namespace regexes of their own.
+        """
+        if self.get_option('only-namespaces'):
+            return [n for n in self.get_option('only-namespaces').split(':')]
+
+        collect_regexes = [
+            r'^openshift\-.+$',
+            r'^kube\-.+$'
+        ]
+
+        if self.get_option('add-namespaces'):
+            for nsp in self.get_option('add-namespaces').split(':'):
+                collect_regexes.append(r'^%s$' % nsp)
+
+        return collect_regexes
+
+    def _reduce_namespace_list(self, nsps):
+        """Reduce the namespace listing returned to just the ones we want to
+        collect from. By default, as requested by OCP support personnel, this
+        must include all 'openshift' prefixed namespaces
+
+            :param nsps list:            Namespace names from oc output
+        """
+
+        def _match_namespace(namespace):
+            """Match a particular namespace for inclusion (or not) in the
+            collection phases
+
+                :param namespace str:   The name of a namespace
+            """
+
+            for regex in self.collect_regexes:
+                if re.match(regex, namespace):
+                    return True
+            return False
+
+        self.collect_regexes = self._setup_namespace_regexes()
+
+        return list(set([n for n in nsps if _match_namespace(n)]))
+
+    def _get_namespaces(self):
+        res = self.exec_cmd(
+            'microshift get namespaces'
+            ' -o custom-columns=NAME:.metadata.name'
+            ' --no-headers')
+        if res['status'] == 0:
+            return self._reduce_namespace_list(res['output'].split('\n'))
+        return []
+
+    def _get_cluster_resources(self):
+        """Get cluster-level (non-namespaced) resources to collect
+        """
+        global_resources = [
+            'apiservices',
+            'certificatesigningrequests',
+            'clusterrolebindings',
+            'clusterroles',
+            'componentstatuses',
+            'csidrivers',
+            'csinodes',
+            'customresourcedefinitions',
+            'flowschemas',
+            'ingressclasses',
+            'logicalvolumes',
+            'mutatingwebhookconfigurations',
+            'nodes',
+            'persistentvolumes',
+            'priorityclasses',
+            'prioritylevelconfigurations',
+            'rangeallocations',
+            'runtimeclasses',
+            'securitycontextconstraints',
+            'selfsubjectaccessreviews',
+            'selfsubjectrulesreviews',
+            'storageclasses',
+            'subjectaccessreviews',
+            'tokenreviews',
+            'validatingwebhookconfigurations',
+            'volumeattachments'
+        ]
+
+        _filtered_resources = []
+
+        for resource in global_resources:
+            res = self.exec_cmd("oc get --kubeconfig %s %s" % (self.get_option('kubeconfig'), resource))
```

Comment:
flake8 is failing here due to line length, need to break this up over multiple lines.

Reply:
Done